src/main/java/graphics/renderer/RenderBatch.java

```diff
@@ -353,4 +283,114 @@ public int zIndex() {
     public int compareTo(RenderBatch a) {
         return Integer.compare(this.zIndex, a.zIndex);
     }
+
+    private void checkFullness() {
+        if (dataOffset >= data.length) {
+            isFull = true;
+            isFull_Textures = false;
+        }
+    }
+
+    /**
+     * Push a float to the data array
+     * @param f the value
+     */
+    public void pushFloat(float f) {

```

Comment:
does it make sense to check, whether the array is full, to prevent issues, or does this slow the whole process down?

Reply:
checkFullness does that

Reply:
but its executing checkFullness after the array access...